sdk/communication/communication-administration/src/communicationIdentity/communicationIdentityClient.ts

```diff
@@ -98,9 +110,18 @@ export class CommunicationIdentityClient {
       }
     };
 
-    const authPolicy = createCommunicationAccessKeyCredentialPolicy(credential);
-    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+    let authPolicy: RequestPolicyFactory;
 
+    if (isTokenCredential(credentialOrOptions)) {
```

Comment:
Can we introduce a new `createCommunicationAuthPolicy` in the common package that accepts `KeyCredential | TokenCredential`? Then we don't have to do the `if`s and `as` type assertions in the clients.

Reply:
Created new helper function for this.